kale/loaddata/multiomics_datasets.py

```diff
@@ -0,0 +1,466 @@
+            present in order to skip downloading.
+        random_split (bool, optional): Whether to split the dataset into random train and test subsets. (default:
+            ``False``)
+        train_size (float, optional): The proportion of the dataset to include in the train split that should be between
+            0.0 and 1.0. This parameter is used when ``random_split`` is ``True``.
+        transform (callable, optional): A function/transform that takes in an array_like data and returns a transformed
+            version. The data object will be transformed before every access. (default: ``None``)
+        pre_transform (callable, optional): A function/transform that takes in an array_like data and returns a
+            transformed version. The data object will be transformed before being saved to disk. (default: ``None``)
+        target_pre_transform (callable, optional): A function/transform that takes in an array_like of labels and
+            returns a transformed version. The label object will be transformed before being saved to disk. (default:
+            ``None``)
+    """
+
+    def __init__(
+        self,
+        root: str,
+        num_modalities: int,
+        num_classes: int,
+        url: str = None,
+        raw_file_names: List[str] = None,
+        random_split: bool = False,
+        train_size: float = 0.7,
+        transform: Optional[Callable] = None,
+        pre_transform: Optional[Callable] = None,
+        target_pre_transform: Optional[Callable] = None,
+    ) -> None:
+        self._url = url
+        self._raw_file_names = raw_file_names
+        self._num_modalities = num_modalities
+        self._num_classes = num_classes
+        self._random_split = random_split
+        self._train_size = train_size
+        self._target_pre_transform = target_pre_transform
+        self._processed_file_names = "data.pt"
+        super().__init__(root, transform, pre_transform)
+
+    @property
+    def raw_file_names(self) -> Optional[List[str]]:
+        r"""The name of the files in the ``self.raw_dir`` folder that must be present in order to skip
+        downloading."""
+        return self._raw_file_names
+
+    @property
+    def processed_file_names(self) -> Union[str, List[str], Tuple]:
+        r"""The name of the files in the ``self.processed_dir`` folder that must be present in order to skip
+        processing."""
+        return self._processed_file_names
+
+    def download(self) -> None:
+        r"""Downloads the dataset to the ``self.raw_dir`` folder."""
+        path = download_url(self._url, self.raw_dir)
+        extract_zip(path, self.raw_dir)
+
+    def process(self) -> None:
+        r"""Processes the dataset to the ``self.processed_dir`` folder. This function reads input files, creates a
+        ''Data'' object, and saves it into the ''processed_dir''."""
+        data_list = []
+
+        if self._random_split:
```

Comment:
The following can be written more compactly by nesting if within the for modality. Talk in person to clarify.

Reply:
Refactored.